src/lib/logical_query_plan/join_node.hpp

```diff
@@ -22,17 +22,21 @@ class JoinNode : public EnableMakeForLQPNode<JoinNode>, public AbstractLQPNode {
   // Constructor for Cross Joins. join_mode has to be JoinMode::Cross
   explicit JoinNode(const JoinMode join_mode);
 
-  // Constructor for predicated joins
+  // TODO(anyone) this constructor is only used for cost estimation in abstract_join_ordering_algorithm
```

Comment:
nap, away with it ;)

Reply:
I want to remove the obsolete JoinNode constructor
```
JoinNode(const JoinMode join_mode, const std::shared_ptr<AbstractExpression>& join_predicate);
```
since we have the new one:
```
JoinNode(const JoinMode join_mode, const std::vector<std::shared_ptr<AbstractExpression>>& join_predicates);
```
Removing the old constructor includes adapting the locations where the old constructor is called.
But what could we do in this case:
```
const auto join_node = JoinNode::make(join_mode, join_predicate);
``` 
Just wrapping the abstract expression `join_predicate` with `{ }` will cause:
```
no matching function for call to ‘opossum::JoinNode::make(const opossum::JoinMode&, <brace-enclosed initializer list>)’
const auto join_node = JoinNode::make(join_mode, {join_predicate});
```

One solution would be the following adaptation:
```
const auto join_node = JoinNode::make(join_mode, std::vector<std::shared_ptr<AbstractExpression>>{join_predicate});
```

But this seems to be a bit ugly and would decrease the readability, wouldn't it?